source/server/connection_handler_impl.cc

```diff
@@ -185,30 +185,41 @@ void ConnectionHandlerImpl::ActiveListener::onAccept(
 }
 
 void ConnectionHandlerImpl::ActiveListener::newConnection(Network::ConnectionSocketPtr&& socket) {
-  Network::ConnectionPtr new_connection = parent_.dispatcher_.createServerConnection(
-      std::move(socket), config_.transportSocketFactory().createTransportSocket());
+  // Find matching filter chain.
+  const auto filter_chain = config_.filterChainManager().findFilterChain(socket);
+  if (filter_chain == nullptr) {
+    ENVOY_LOG_TO_LOGGER(parent_.logger_, debug,
+                        "closing connection: no matching filter chain found");
+    socket->close();
```

Comment:
Should we bump a stat or something else that could be monitored? Seems a debug log here is effectively a silent fail in production. I realize this is existing behavior, so a TODO is fine as well.

Reply:
Done.